endpoints/openrtb2/auction_test.go

```diff
@@ -2912,6 +2912,108 @@ func TestValidateNativeAssetData(t *testing.T) {
 	}
 }
 
+func TestFPDHoldAuction(t *testing.T) {
+
+	deps := &endpointDeps{
+		fakeUUIDGenerator{},
+		&warningsCheckExchange{},
+		newParamsValidator(t),
+		&mockStoredReqFetcher{},
+		empty_fetcher.EmptyFetcher{},
+		empty_fetcher.EmptyFetcher{},
+		nil,
+		&metricsConfig.DummyMetricsEngine{},
+		analyticsConf.NewPBSAnalytics(&config.Analytics{}),
+		map[string]string{},
+		false,
+		[]byte{},
+		openrtb_ext.BuildBidderMap(),
+		nil,
+		nil,
+		hardcodedResponseIPValidator{response: true},
+	}
+
+	if specFiles, err := ioutil.ReadDir("../../firstpartydata/tests/holdauction"); err == nil {
+		for _, specFile := range specFiles {
+			fileName := "../../firstpartydata/tests/holdauction/" + specFile.Name()
+			fpdFile, err := loadFpdFile(fileName)
+
+			if err != nil {
+				t.Errorf("Unable to load file: %s", fileName)
+			}
+
+			var resRequest openrtb2.BidRequest
+			err = json.Unmarshal(fpdFile.ResultRequestData, &resRequest)
+			if err != nil {
+				t.Errorf("Unable to unmarshal input request: %s", fileName)
+			}
+
+			reqBody := string(fpdFile.InputRequestData)
+			deps.cfg = &config.Configuration{MaxRequestSize: int64(len(reqBody))}
+
+			req := httptest.NewRequest("POST", "/openrtb2/auction", strings.NewReader(reqBody))
+			recorder := httptest.NewRecorder()
+
+			deps.Auction(recorder, req, nil)
+
+			if recorder.Code != http.StatusOK {
+
+				resp, err := ioutil.ReadAll(recorder.Body)
+				assert.NoError(t, err, "Error shuld not be nil")
+
+				for i := range fpdFile.ValidationErrors {
+					assert.Contains(t, string(resp), fpdFile.ValidationErrors[i], "Incorrect first party data warning message")
+				}
+				continue
```

Comment:
Why do we need a `continue` here?

Reply:
In case we expect validation errors we will check error only and will not check other request data in further code. We need to go to next iteration in this case.